subsys/bluetooth/audio/ascs.c

```diff
@@ -507,6 +508,10 @@ static void ascs_cp_rsp_add(uint8_t id, uint8_t op, uint8_t code,
 	 */
 	case BT_ASCS_RSP_NOT_SUPPORTED:
 	case BT_ASCS_RSP_TRUNCATED:
+		if (rsp->num_ase) {
+			net_buf_simple_remove_mem(&rsp_buf,
+						  sizeof(*ase_rsp) * rsp->num_ase);
+		}
```

Comment:
Hmm, I am not so sure about that. Are you sure we have so much data in the buffer?

Reply:
It looks OK I guess, but could you add an __ASSERT here to make sure the size matches buffer length?
I guess it should look like
```
__ASSERT_NO_MSG(rsp_buf.len == (sizeof(*rsp) + sizeof(*ase_rsp) * rsp->num_ase));
```